### ISSUE TITLE
Working without composer

### ISSUE THREAD

**Issue by vicentor** [AlexaCRM/php-crm-toolkit#18]
Without composer, we tried following statement:
In `/examples/` you can find a few examples of toolkit usage. Copy `config.example.php` to `config.php`, set up credentials for your CRM and you are ready to go!

Doing so ends up with the following error:
Fatal error: Class 'Psr\Log\NullLogger' not found in /home/magikweb/public_html/dev/d365/oauth2/php-crm-toolkit/src/Client.php on line 137

I am not familiar with either composer or namespaces in general, but maybe there's a lacking inclusion assuming we're not using the autoloader?

**Comment by vicentor** [comment]
Actual script: https://dev.magikweb.ca/d365/oauth2/php-crm-toolkit/examples/cli-getallcontacts.php

**Comment by wizardist** [comment]
Hi @vicentor 

Composer is the preferred way to get the toolkit up and running, but you can live without it, although some effort would be needed.

You would need to grab all required libraries (which is only [psr/log](https://github.com/php-fig/log) for now) and write a custom autoloader. See http://php.net/manual/en/function.spl-autoload-register.php

An example of such autoloader can be found in the `init.php` of the toolkit. You need to specify the correct `$namespacePrefix` (which is `Psr\\Log`), and `$baseDirectory` (which is wherever you place the psr/log library).

**Comment by vicentor** [comment]
Thank you for the follow-up @wizardist, you were very clear and it all makes sense.
We successfully got the files through Composer and uploaded them onto our server.

We get this error;
Fatal error: Class 'AlexaCRM\CRMToolkit\Settings' not found in /home/magikweb/public_html/dev/d365/oauth2/index.php on line 15

When trying the main example on the Github readme.md, could it be that the class `Settings` is not mapped properly by default?

Code:
![settings](https://cloud.githubusercontent.com/assets/2099885/24162303/7343c04a-0e3d-11e7-88dc-e35fdd6051bb.png)

Structure:
![file-structure](https://cloud.githubusercontent.com/assets/2099885/24162315/7889a506-0e3d-11e7-9999-184f852031a1.png)

I apologize if this is obvious details, we're trying to make it work "out-of-the-box" with simple calls before progressing.

**Comment by wizardist** [comment]
@vicentor What's in your `vendor` directory? If `alexacrm/php-crm-toolkit` is there, then it's odd, and I try to recreate.

Did you install the library by executing `composer require alexacrm/php-crm-toolkit`?

**Comment by vicentor** [comment]
You are right! I used `composer install` after downloading manually the `composer.json` file. That gave me the wrong files I believe.

![files-wrong](https://cloud.githubusercontent.com/assets/2099885/24199956/d7ec04d6-0ee1-11e7-9e7e-8ea4671fe146.png)

I tried using `composer require alexacrm/php-crm-toolkit` only and now have the folder you mentioned.

![files-right](https://cloud.githubusercontent.com/assets/2099885/24199986/ec599366-0ee1-11e7-9bf6-3ffdee11fd32.png)

I appreciate your help, I am aware that at this point the real issue is myself.
